README.md

```diff
@@ -1 +1,192 @@
+
+How can we test for this? Test it again! Realistically, our tests should always test our functions with many different parameters, expecting correct answers from our functions. That way, if we have a test adding `1` and `3` (to equal `4`), and also adding `50` and `24329` (to equal `24379`), we can, with some confidence, know that our number adding function actually works as we expect.
+
+Let's modify our first test:
+
+```js
+describe('Basic Functionality', function () {
+  it('should add two numbers', function () {
+    expect(addNumbers(1, 3)).toBe(4);
+    expect(addNumbers(343, 9283)).toBe(9626);
+    expect(addNumbers(1223, 21)).toBe(1244);
+    expect(addNumbers(10, 653)).toBe(663);
+  });
+});
+```
+
+Okay, now we've got multiple test cases, let's run our tests again. As expected, our first one passed (as our number adding function is only returning `4`), but the others have failed. Now we need to head back to our number adder and actually implement it correctly.
+
+```js
+function addNumbers(numberOne, numberTwo) {
+  return numberOne + numberTwo;
+}
+```
+
+We run our tests again and they all pass - happy days! The code goes into production, but two days later we receive a bug report saying that when people are adding `2` and `2` together, they get `22` instead of `4`. Oh no, what could've gone wrong?
+
+Our function doesn't support strings! In JavaScript, if we were to do `"2" + 2`, we'd get `22` as the result because the `+` operator adds two strings together, as well as actually doing addition between two numbers.
+
+Let's head back to our tests, and add some use-cases for when the function gets called with parameters that don't match what it expects.
+
+```js
+describe('Basic Functionality', function () {
+  it('should add two numbers', function () {
+    expect(addNumbers(1, 3)).toBe(4);
+    expect(addNumbers(343, 9283)).toBe(9626);
+    expect(addNumbers(1223, 21)).toBe(1244);
+    expect(addNumbers(10, 653)).toBe(663);
+    expect(addNumbers('99', 1)).toBe(100);
+    expect(addNumbers('23', '59')).toBe(82);
+  });
+});
+```
+
+If we run our tests again, they will fail (as we expect). Now, we can head back to our code and make amendments to handle these use cases.
+ 
+```js
+function addNumbers(numberOne, numberTwo) {
+  return parseFloat(numberOne, 10) + parseFloat(numberTwo, 10);
+}
+```
+
+Brilliant. Our tests run and pass and we have a well coded, tested piece of code. We can then make changes to this code further, running our tests against it along the way, ensuring no previous expected functionality has been broken.
+
+## Jasmine and Karma
+
+To do TDD, we need to install Jasmine and Karma. Clone this repo completely.
+
+Open your terminal, and run
+
+```bash
+sudo npm install -g karma-cli
+```
+
+This will install the Karma command line interface tool, so we can run tests using `karma`.
+
+We also need a copy of Karma for our project, as well as Jasmine too.
```

Comment:
can you briefly touch on the difference between installing karma vs install karma for our projects?